File: MyProject/Source/MyProject/Private/BasicFrame/Libs/EventHandle/ResEventDispatch.cpp
#include "MyProject.h"
#include "ResEventDispatch.h"
#include "IDispatchObject.h"

ResEventDispatch::ResEventDispatch()
{

}

void ResEventDispatch::dispatchEvent(IDispatchObject* dispatchObject)
{
	EventDispatch::dispatchEvent(dispatchObject);
	clearEventHandle();
}

File: Project/MyProject/MyProject/Content/MyAsset/Lua/MyLua/Libs/EventHandle/ResEventDispatch.lua
MLoader("MyLua.Libs.Core.GlobalNS");
MLoader("MyLua.Libs.Core.Class");
MLoader("MyLua.Libs.EventHandle.EventDispatch");

local M = GlobalNS.Class(GlobalNS.EventDispatch);
M.clsName = "ResEventDispatch";
GlobalNS[M.clsName] = M;

function M:ctor()
    
end

function M:dtor()

end

function M:dispatchEvent(dispatchObject)
    M.super.dispatchEvent(self, dispatchObject);
	
    self:clearEventHandle();
end

return M;